core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -359,6 +354,11 @@ class KafkaServer(
         val brokerInfo = createBrokerInfo
         val brokerEpoch = zkClient.registerBroker(brokerInfo)
 
+        lifecycleManager = new BrokerLifecycleManager(config,
+          time,
+          threadNamePrefix,
+          zkBrokerEpoch = Some(brokerEpoch))
```

Comment:
Won't this value change every while the broker is alive? I mean won't the broker be able to re-register/establish new session with Zookeeper after losing it?

Reply:
Good catch. We do get new broker epochs after Zk session re-initializes. I'll change this to use the same pattern we use in AlterPartitionManager